core/breakpoints/manager.php

```diff
@@ -0,0 +1,330 @@
+<?php
+namespace Elementor\Core\Breakpoints;
+
+use Elementor\Core\Base\Module;
+use Elementor\Core\Kits\Documents\Tabs\Settings_Layout;
+use Elementor\Core\Responsive\Files\Frontend;
+use Elementor\Core\Responsive\Responsive;
+use Elementor\Plugin;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Manager extends Module {
+
+	const BREAKPOINT_OPTION_PREFIX = 'viewport_';
+	const BREAKPOINT_KEY_MOBILE = 'mobile';
+	const BREAKPOINT_KEY_MOBILE_EXTRA = 'mobile_extra';
+	const BREAKPOINT_KEY_TABLET = 'tablet';
+	const BREAKPOINT_KEY_TABLET_EXTRA = 'tablet_extra';
+	const BREAKPOINT_KEY_LAPTOP = 'laptop';
+	const BREAKPOINT_KEY_DESKTOP = 'desktop';
+	const BREAKPOINT_KEY_WIDESCREEN = 'widescreen';
+
+	/**
+	 * Legacy breakpoints.
+	 *
+	 * For Backwards compatibility, Holds the old responsive breakpoints.
+	 *
+	 * @since 3.1.0
+	 * @access private
+	 * @static
+	 *
+	 * @var array Legacy breakpoints.
+	 */
+	private $legacy_breakpoints = [
+		'xs' => 0,
+		'sm' => 480,
+		'md' => 768,
+		'lg' => 1025,
+		'xl' => 1440,
+		'xxl' => 1600,
+	];
+
+	private $config;
+	private $breakpoints;
+
+	public function get_name() {
+		return 'breakpoints';
+	}
+
+	/**
+	 * Get Breakpoints
+	 *
+	 * Retrieve the array of --enabled-- breakpoints, or a single breakpoint if a name is passed.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param $breakpoint_name
+	 * @return mixed
+	 */
+	public function get_breakpoints( $breakpoint_name = null ) {
+		if ( ! $this->breakpoints ) {
+			$this->breakpoints = $this->init_breakpoints();
+		}
+		return self::get_items( $this->breakpoints, $breakpoint_name );
+	}
+
+	/**
+	 * Get Legacy Breakpoints
+	 *
+	 * Retrieve the array of legacy breakpoints. Used for backwards compatibility.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return array $legacy_breakpoints
+	 */
+	public function get_legacy_breakpoints() {
+		return $this->legacy_breakpoints;
+	}
+
+	/**
+	 * Get Active Breakpoints with Previous Values
+	 *
+	 * Returns an array of breakpoints with the breakpoint names as keys, and the previous breakpoint's value as their
+	 * value. This is used by the CSS generation classes to get the minimum points of each breakpoint. The returned
+	 * array includes all breakpoints EXCEPT for mobile, since mobile is the lowest breakpoint and as such, it has
+	 * a minimum point of 0px.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return array
+	 */
+	public function get_active_breakpoints_with_previous_values() {
```

Comment:
Check if we can adjust the `Stylesheet` to work without previous values

Reply:
Done.